plugins/tangle/solidifier_gossip.go

```diff
@@ -70,19 +76,26 @@ func checkSolidityAndPropagate(transaction *hornet.Transaction) {
 
 			solid, _ := checkSolidity(tx, true)
 			if solid {
-				if int32(time.Now().Unix())-tx.GetSolidificationTimestamp() > solidifierThresholdInSeconds {
+				if int32(time.Now().Unix())-tx.GetTransaction().GetSolidificationTimestamp() > solidifierThresholdInSeconds {
 					// Skip older transactions
+					tx.Release() //-1
 					continue
 				}
 
 				transactionApprovers, _ := tangle.GetApprovers(txHash)
 				for _, approverHash := range transactionApprovers.GetHashes() {
-					approver, _ := tangle.GetTransaction(approverHash)
-					if approver != nil {
-						txsToCheck[approverHash] = approver
+					approver := tangle.GetCachedTransaction(approverHash) //+1
```

Comment:
I think there is a `Release` missing if the approver doesn't exist

Reply:
Good one 👍